client/scss/components/_footer.scss

```diff
@@ -85,12 +85,34 @@
   width: 100%;
 
   @include media-breakpoint-up(sm) {
-    width: 250px;
+    width: theme('spacing.64');
   }
 
   &--primary {
+    // Support basic layout support for items added via `extra_footer_actions`
+
+    align-items: stretch;
+    display: grid;
+    gap: theme('spacing.2');
+    grid-auto-flow: column;
+
+    > .button {
+      // There are default .button styles that would:
+      // - set height responsively
+      // - add a margin when there is a .button sibling
+      // Unset these styles in favor of a fixed height and the grid gap
+      height: $text-input-height;
```

Comment:
Tested a few different scenarios and this seems OK

<img width="558" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/a6819a38-61ed-4bd1-bf27-d9a84338cc60">

<img width="558" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/f9dbad46-bcf6-435a-92a1-0774bad64ce9">

<img width="628" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/c6a39e67-b69b-4186-aef2-9fd485583159">

<img width="543" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/c41776d0-1480-4f0c-a691-cb696da2da85">

Reply:
what about cases where one wraps that `.button` in a form? wagtail-localize used to do that for the additional buttons (a la "import from local file")

Reply:
Hmm, OK, good point. I wanted to remove the `>` but was afraid it would be risky.

The main dropdown action button from Wagtail is also wrapped inside another element, and without removing the `>` this technically still uses the default styles from `.button` (see bottom right), but the `_dropdown-button.scss` has its own `min-height` styles to handle the case for smaller screens so that it uses this same height instead of the smaller one:

<img width="998" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/a5195316-477b-4487-bae7-a8857641d441">

I think it should be safe to remove the `>`.

Reply:
Tried wrapping it in a `<form>` but the browser won't render it, probably because the footer actions are inside the page editor's form, and nested forms aren't valid HTML5.

Here's with a `<div>` instead (and `> .button` changed to `.button`)

<img width="1800" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/c16e71f9-98f1-4438-b5f0-d6947c6a11a3">



<img width="1276" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/2e89b54e-c5eb-4335-89e0-db7a70340281">

Reply:
Thanks for checking!

localize is a bit special in that it replaces the edit form with its React bits (one more reason to rewrite with good old panels).

Reply:
Thanks everyone. A great solution in the end.

Reply:
Team work! Thank you @lb- and @laymonage ❤️